analyzers/tests/SonarAnalyzer.Test/TestCases/SpecifyRouteAttribute.cs

```diff
@@ -0,0 +1,111 @@
+//           ^^^^^^^^^^^^^^^^^^^^^^^^^^
+{
+    private const string ConstantRoute = "ConstantRoute";
+
+    [HttpGet("GetObject")]
+    public IActionResult Get() => View();
+//                       ^^^ Secondary [controller]
+
+    [HttpGet("GetFirst")]
+    [HttpGet("GetSecond")]
+    public IActionResult GetMultipleTemplates() => View();  // Secondary [controller]
+
+    [HttpGet("GetFirst")]
+    [HttpPut("GetSecond")]
+    public IActionResult Mix() => View();                   // Secondary [controller]
+
+    [HttpPost("CreateObject")]
+    public IActionResult Post() => View();                  // Secondary [controller]
+
+    [HttpPut("UpdateObject")]
+    public IActionResult Put() => View();                   // Secondary [controller]
+
+    [HttpDelete("DeleteObject")]
+    public IActionResult Delete() => View();                // Secondary [controller]
+
+    [HttpPatch("PatchObject")]
+    public IActionResult Patch() => View();                 // Secondary [controller]
+
+    [HttpHead("Head")]
+    public IActionResult HttpHead() => View();              // Secondary [controller]
+
+    [HttpOptions("Options")]
+    public IActionResult HttpOptions() => View();           // Secondary [controller]
+
+    [Route("details")]
+    public IActionResult GetDetails() => View();            // FN
+
+    [HttpGet("details", Order = 1)]
+    public IActionResult GetDetails2() => View();           // Secondary [controller]
+
+    [HttpGet("details", Order = 1, Name = "Details")]
+    public IActionResult GetDetails3() => View();           // Secondary [controller]
+
+    [HttpGet(ConstantRoute)]
+    public IActionResult GetDetails4() => View();           // Secondary [controller]
+
+    [HttpGet("""
+             ConstantRoute
+             """)]
+    public IActionResult GetDetails5() => View();           // Secondary [controller]
+
+    [HttpGet($"Route {ConstantRoute}")]
+    public IActionResult GetDetails6() => View();           // Secondary [controller]
+
+    [HttpGet($"""
+             {ConstantRoute}
+             """)]
+    public IActionResult GetDetails7() => View();           // Secondary [controller]
+
+    [HttpGet("GetObject")]
+    public ActionResult GetDetails8() => View();            // Secondary [controller]
+
+    // [HttpPost("Comment")]
+    public IActionResult Comment() => View();
+}
+
+public class WithUserDefinedAttribute : Controller          // Noncompliant [customAttribute]
+{
+    [MyHttpMethod("Test")]
+    public IActionResult Index() => View();                 // Secondary [customAttribute]
+
+    private sealed class MyHttpMethodAttribute(string template) : HttpMethodAttribute([template]) { }
+}
+
+public class WithCustomGetAttribute : Controller
+{
+    [HttpGet("Test")]                                       // Compliant, behavior is user-defined
+    public IActionResult Index() => View();
+    private sealed class HttpGetAttribute(string template) : HttpMethodAttribute([template]) { }
```

Comment:
In `WithUserDefinedAttribute` we have the property and the class separated by a space, whereas here they are not.
I would be consistent and have a space there (as per our production code styling rules).